sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1092,13 +1082,30 @@ object CodeGenerator extends Logging {
         logInfo(s"\n${CodeFormatter.format(code, maxLines)}")
         throw new CompileException(msg, e.getLocation)
     }
+
+    // Check if compiled code has a too large function
+    methodsToByteCodeSize.foreach { case (name, byteCodeSize) =>
+      if (byteCodeSize > SQLConf.get.hugeMethodLimit) {
+        val clazzName = evaluator.getClazz.getSimpleName
+        val methodName = name.replace("$", "")
+        val msg = s"failed to compile: the size of $clazzName.$methodName was $byteCodeSize and " +
+          "this value went over the limit `spark.sql.codegen.hugeMethodLimit`" +
+          s"(${SQLConf.get.hugeMethodLimit}). To avoid this error, you can make this limit higher."
+        logError(msg)
+        val maxLines = SQLConf.get.loggingMaxLinesForCodegen
+        logInfo(s"\n${CodeFormatter.format(code, maxLines)}")
+        throw new CompileException(msg, null)
```

Comment:
Previously, it works although it does not perform well. Now, after this change, the query does not work. This could causes regressions. I think we should make a change to automatically fall back to the slow mode.

Reply:
ok, I try to refactor this.

Reply:
We should turn back to non-codegen execution for too long function.

Reply:
yea, I'll do so. Thanks!